src/basic/audit-util.c

```diff
@@ -80,7 +123,18 @@ bool use_audit(void) {
                         if (!cached_use)
                                 log_debug_errno(errno, "Won't talk to audit: %m");
                 } else {
-                        cached_use = true;
+                        /* If we try and use the audit fd but get ECONNREFUSED, it is because
+                         * we are not in the initial user namespace, and the kernel does not
+                         * have support for audit outside of the initial user namespace. */
+                        int r = try_audit_request(fd);
```

Comment:
We treat `r` as a special variable, and it shall generally be placed at the beginning of the function.